.config/doom/config.org
#+TITLE: Config
:PROPERTIES:
#+AUTHOR: Debarghya Datta
#+DATE: Thu Jan 27 18:44:35 2022
#+DESCRIPTION: Personal Dotfiles
#+OPTIONS:  timestamp: nil toc:1 num:nil reveal_slide_number:nil
#+REVEAL_TRANS: concave
#+REVEAL_THEME: serif
#+PROPERTY: header-args:emacs-lisp :tangle config.el :noweb
#+STARTUP: hideblocks fold
:END:

* :: :toc:
- [[#my-doom-emacs-config][My Doom Emacs config]]
- [[#personal-info][Personal Info🙍]]
- [[#fonts-and-stuff][Fonts and stuff📎]]
- [[#theming][Theming🌟]]
- [[#dired][Dired📁]]
- [[#editor-customization][Editor customization📋]]
- [[#org-mode-customization][Org mode customization🦄]]
  - [[#set-font-sizes-for-each-header-level-in-org][Set font sizes for each header level in Org]]
  - [[#source-code-buffer][Source code buffer]]
  - [[#revealjs-init][Reveal.js init🖨]]
- [[#zen-mode][Zen Mode]]

* My Doom Emacs config
My personal doom emacs config.

* Personal Info🙍
Here i provide the information that is required for ssl or different clients(git, rss, etc.)
#+begin_src emacs-lisp
(setq user-full-name "Debarghya Datta"
      user-mail-address "debarghya.datta89@gmail.com")
#+end_src

* Fonts and stuff📎
Here we can set 5 different types of fonts, and can use them according to our use cases
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "monospace" :size 19)
      doom-big-font (font-spec :family "Hack Nerd Font" :size 25)
      doom-variable-pitch (font-spec :family "ETBembo" :size 22))
#+end_src
Please check the "ETBembo" font in the github.

* Theming🌟
We can theme our emacs with a single line of elisp
press *spc+h+t* to see all the themes available in doom emacs

#+begin_src emacs-lisp
(defun synchronize-theme ()
  (let* ((light-theme 'doom-one)
         (dark-theme 'doom-moonlight)
         (start-time-light-theme 6)
         (end-time-light-theme 18)
         (hour (string-to-number (substring (current-time-string) 11 13)))
         (next-theme (if (member hour (number-sequence start-time-light-theme end-time-light-theme))
                         light-theme dark-theme)))
    (when (not (equal doom-theme next-theme))
      (setq doom-theme next-theme))))

(synchronize-theme)
#+end_src

My favourites are doom-one, doom-nord, doom-monokai-pro

* Dired📁
Proper association of file extensions
#+begin_src emacs-lisp
(setq dired-open-extensions '(("gif" . "sxiv")
                              ("jpg" . "sxiv")
                              ("png" . "sxiv")
                              ("mkv" . "mpv")
                              ("mp4" . "mpv")))
#+end_src

* Editor customization📋
We set the line number display, I prefer relative line numbering, but can be changed to default by setting it as 't'
#+begin_src emacs-lisp
(setq display-line-numbers-type 'relative)
(whitespace-mode t)
#+end_src

* Org mode customization🦄

Here we set the deafult location for all the org files in ur system
#+begin_src emacs-lisp
(after! org
    (setq org-directory "~/dox/org/"
          org-agenda-files '("~/dox/org/agenda.org")
          org-ellipsis "  "
          org-cycle-separator-lines 2
          org-pretty-entities t
          org-startup-with-inline-images t
          org-image-actual-width '(300)
          org-src-window-setup 'current-window
          org-superstar-headline-bullets-list '(" ")
          ;; org-superstar-headline-bullets-list '("◉" "●" "○" "◆" "●" "○" "◆")
          ;; org-superstar-item-bullet-alist '((?+ . ?➤) (?- . ?✦))
          org-log-done 'time
          org-hide-emphasis-markers t
          org-link-abbrev-alist
          '(("google" . "http://www.google.com/search?q=")
            ("arch-wiki" . "https://wiki.archlinux.org/index.php/"))
          org-todo-keywords
              '((sequence
              "TODO(t)"
              "PROJ(p)"
              "STUDY(s)"
              "WAIT(w)"
              "|"
              "DONE(d)"
              "CANCELLED(c)" )))
    (set-popup-rule! "^\\*Org Src" :ignore t))
#+end_src

** Set font sizes for each header level in Org

Set the Org heading levels to be different font sizes.  So I choose to have level 1 headings to be 140% in height, level 2 to be 130%, etc.  Other interesting things you could play with include adding :foreground color and/or :background color if you want to override the theme colors.
#+begin_src emacs-lisp
(add-hook 'org-mode-hook (lambda ()
                           (+zen/toggle +1)
                           (setq display-line-numbers nil)
                           (variable-pitch-mode +1)
                           ))
#+end_src
Experiment with it

** Source code buffer

For editing source code use the current buffer.
#+begin_src emacs-lisp
(setq org-src-window-setup 'current-window)
#+end_src

** Reveal.js init🖨

Reveal,js is a really interesting tool, which can create sick looking slides from markdown.
We will use org in place of markdown(ofcourse!)
#+begin_src emacs-lisp
(setq org-re-reveal-root
      "file:///home/devildev/dox/reveal.js")
(setq org-re-reveal-plugins '(highlight))
(setq org-re-reveal-highlight-css
      "file:///home/devildev/dox/reveal.js/plugin/highlight/monokai.css")
#+end_src

* Zen Mode

Doom zooms to much in zen mode.
#+begin_src emacs-lisp
(setq +zen-text-scale 0.8)
(defvar mixed-pitch-modes '(org-mode LaTeX-mode markdown-mode gfm-mode Info-mode)
  "Modes that `mixed-pitch-mode' should be enabled in, but only after UI initialisation.")
(defun init-mixed-pitch-h ()
  "Hook `mixed-pitch-mode' into each mode in `mixed-pitch-modes'.
Also immediately enables `mixed-pitch-modes' if currently in one of the modes."
  (when (memq major-mode mixed-pitch-modes)
    (mixed-pitch-mode 1))
  (dolist (hook mixed-pitch-modes)
    (add-hook (intern (concat (symbol-name hook) "-hook")) #'mixed-pitch-mode)))
(add-hook 'doom-init-ui-hook #'init-mixed-pitch-h)
#+end_src

* Misc :noexport:

- _Tasks_
  - [ ] =Document it properly=
  - [-] =Clean the unnecesary code=

Auto tangle the file after save

;; Local Variables:
;; eval: (add-hook 'after-save-hook (lambda ()(if (y-or-n-p "Tangle?")(org-babel-tangle))) nil t)
;; End:


.config/doom/config.el
(setq user-full-name "Debarghya Datta"
      user-mail-address "debarghya.datta89@gmail.com")

(setq doom-font (font-spec :family "monospace" :size 19)
      doom-big-font (font-spec :family "Hack Nerd Font" :size 25)
      doom-variable-pitch (font-spec :family "ETBembo" :size 22))

(defun synchronize-theme ()
  (let* ((light-theme 'doom-one)
         (dark-theme 'doom-moonlight)
         (start-time-light-theme 6)
         (end-time-light-theme 18)
         (hour (string-to-number (substring (current-time-string) 11 13)))
         (next-theme (if (member hour (number-sequence start-time-light-theme end-time-light-theme))
                         light-theme dark-theme)))
    (when (not (equal doom-theme next-theme))
      (setq doom-theme next-theme))))

(synchronize-theme)

(setq dired-open-extensions '(("gif" . "sxiv")
                              ("jpg" . "sxiv")
                              ("png" . "sxiv")
                              ("mkv" . "mpv")
                              ("mp4" . "mpv")))

(setq display-line-numbers-type 'relative)
(whitespace-mode t)

(after! org
    (setq org-directory "~/dox/org/"
          org-agenda-files '("~/dox/org/agenda.org")
          org-ellipsis "  "
          org-cycle-separator-lines 2
          org-pretty-entities t
          org-startup-with-inline-images t
          org-image-actual-width '(300)
          org-src-window-setup 'current-window
          org-superstar-headline-bullets-list '(" ")
          ;; org-superstar-headline-bullets-list '("◉" "●" "○" "◆" "●" "○" "◆")
          ;; org-superstar-item-bullet-alist '((?+ . ?➤) (?- . ?✦))
          org-log-done 'time
          org-hide-emphasis-markers t
          org-link-abbrev-alist
          '(("google" . "http://www.google.com/search?q=")
            ("arch-wiki" . "https://wiki.archlinux.org/index.php/"))
          org-todo-keywords
              '((sequence
              "TODO(t)"
              "PROJ(p)"
              "STUDY(s)"
              "WAIT(w)"
              "|"
              "DONE(d)"
              "CANCELLED(c)" )))
    (set-popup-rule! "^\\*Org Src" :ignore t))

(add-hook 'org-mode-hook (lambda ()
                           (+zen/toggle +1)
                           (setq display-line-numbers nil)
                           (variable-pitch-mode +1)
                           ))

(setq org-src-window-setup 'current-window)

(setq org-re-reveal-root
      "file:///home/devildev/dox/reveal.js")
(setq org-re-reveal-plugins '(highlight))
(setq org-re-reveal-highlight-css
      "file:///home/devildev/dox/reveal.js/plugin/highlight/monokai.css")

(setq +zen-text-scale 0.8)
(defvar mixed-pitch-modes '(org-mode LaTeX-mode markdown-mode gfm-mode Info-mode)
  "Modes that `mixed-pitch-mode' should be enabled in, but only after UI initialisation.")
(defun init-mixed-pitch-h ()
  "Hook `mixed-pitch-mode' into each mode in `mixed-pitch-modes'.
Also immediately enables `mixed-pitch-modes' if currently in one of the modes."
  (when (memq major-mode mixed-pitch-modes)
    (mixed-pitch-mode 1))
  (dolist (hook mixed-pitch-modes)
    (add-hook (intern (concat (symbol-name hook) "-hook")) #'mixed-pitch-mode)))
(add-hook 'doom-init-ui-hook #'init-mixed-pitch-h)
